### PR TITLE
Correct letter case of MinGW and MinGW-w64

### PR DESCRIPTION
**Pull request by favonia**
My perfectionism strikes again. :sweat_smile:

1. Facts: MinGW-w64 is a project that supports both 32-bit and 64-bit Windows. It's better than the original MinGW even on 32-bit Windows.
2. I left `Changes` mostly unchanged, focusing on the future.
3. ~~Maybe all the error messages in `configure.ac` should display `MinGW-w64` instead of `MinGW` because almost no one is using the original MinGW. However, maybe someone is still building OCaml compilers with it? (Why?)~~ Done changing them.

Only one word in the standard library is changed. So I suppose this does not require a new entry in `Changes`.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -715,7 +715,7 @@ AS_CASE([$host],
   [*-*-mingw32],
     [AS_CASE([$ocaml_cv_cc_vendor],
       [gcc-[[01234]]-*],
-        [AC_MSG_ERROR(m4_normalize([This version of Mingw GCC is too old.
+        [AC_MSG_ERROR(m4_normalize([This version of MinGW GCC is too old.
```
Comment (dra27) [inline configure.ac]: This, the one on L728 and the reference in sys.mli can also be `MinGW-w64`, if you wish

**Review by dra27 — COMMENTED** [review]
My kind of perfectionism, thanks 🙂

As far as I can tell, this is being more fastidious about the case than the MinGW-w64 project page or Wikipedia pages are being, but that's no reason not to be correct!

IIRC it wasn't just that we never supported building with the mingw.org compilers directly, I think it didn't work - the OCaml _had_ to be compiled with the gcc-mingw package in Cygwin (accessed via `-mno-cygwin` flag to GCC) and GCC 5 was never packaged with that, so I don't think there's any chance it's in use!

File: configure.ac (hunk 1 above)
Comment (favonia) [reply configure.ac]: @dra27 All done!

**Comment by dra27** [comment]
Thanks, @favonia! Incidentally, just for info:

> I did not run autoconf because my autoconf is too new

If you're using a Debian/Ubuntu system, the `autoconf2.69` package provides the `autoconf2.69` command (on Windows, the Cygwin `autoconf2.5` package provides `autoconf-2.69` - it gets installed as part of the `autoconf` package, though). Armed with that, you can then run `tools/autogen autoconf2.69` (or, annoyingly, `tools/autogen autoconf-2.69` on Cygwin) to regenerate `configure` using the older version.

(it's also possible to use `docker` to run an older Ubuntu LTS)

**Pull request merged**